packages/block-library/src/navigation-link/editor.scss

```diff
@@ -50,10 +50,6 @@
 	.wp-block-navigation-link__submenu-icon {
 		margin-left: 4px;
 	}
-
-	.block-editor-rich-text__editable.is-selected:not(.keep-placeholder-on-focus):not(:focus) [data-rich-text-placeholder]::after {
-		display: inline-block;
```

Comment:
Tested and now the placeholder looks good when the item is selected but not focussed:

<img width="481" alt="Screenshot 2020-01-23 at 11 04 14" src="https://user-images.githubusercontent.com/4710635/72974874-1c915580-3dd0-11ea-88c7-2df2daf3da40.png">

Reply:
An issue seems to be still there. If you focus the label, it doesn't show the caret, making the behavior weird to understand.

![image](https://user-images.githubusercontent.com/77539/73686199-60137b80-467c-11ea-8554-076dd28eec4b.png)

Reply:
just in case fixed https://github.com/WordPress/gutenberg/pull/20075 (feel free to review it)